srcpkgs/hg-git/patches/fix-egrep-test-errors.patch

```diff
@@ -0,0 +1,113 @@
+--- a/tests/test-conflict-1.t
```

Comment:
I don't want to carry around `srcpkgs/hg-git/patches/fix-egrep-test-errors.patch` as a patch. It just changes `egrep` to `grep -E` I guess to avoid the GNU "egrep is depreciated" warning?

Could be an alias better, or a `sed` patch in a `pre` step.

Or merged upstream! That'd be nice. This problem is not unique to void, so we should not be carrying this patch.

Reply:
`chroot-grep-3.8_1       /usr/bin/egrep`

I will accept a PR that patches chroot-grep too.

Reply:
> grep 3.8 prints a warning to stderr that could potentially break some builds and the grep package patches it out, but I ran a world rebuild with chroot-grep 3.8 and everything built, so I didn't add the patch here.

From the chroot-grep update PR - is it okay to create a new pr for this?

Reply:
Yes

Reply:
Link it to me on IRC or here and I'll be happy to merge

Reply:
https://github.com/void-linux/void-packages/pull/41593

Reply:
Is this a hard error or just warnings? In every single other package, it's just a harmless warning.

Reply:
The tests work by comparing stdout with the expected result. Since the warnings get posted in stdout, it no longer matches the expected value. Last time I [updated](https://github.com/void-linux/void-packages/commit/e851fe63829a092a79798cc4f355398e73d0fc5f) it, a patch was needed for the tests due to mismatched git version from upstream. It got [fixed upstream](https://foss.heptapod.net/mercurial/hg-git/-/commit/5277b791c8e79568832785560cac491d72044534) in this release when their git version caught up to ours. So I expect this to will be fixed when they move to grep-3.8.

Reply:
The warnings from grep go to stderr, so they are not an issue in most cases.

Reply:
Okay, then they are tracking the stderr as well. All the test fails the same way:
```patch
--- /builddir/hg-git-1.0.1/tests/test-renames.t
+++ /builddir/hg-git-1.0.1/tests/test-renames.t.err
@@ -40,6 +40,7 @@
   $ clonefilt='s/Cloning into/Initialized empty Git repository in/;s/in .*/in .../'
 
   $ git submodule add ../gitsubmodule 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  egrep: warning: egrep is obsolescent; using grep -E
   Initialized empty Git repository in ...
   
   $ fn_git_commit -m 'add submodule'
@@ -73,6 +74,7 @@
 
   $ git mv gamma gamma-new
   $ git submodule add ../gitsubmodule gamma 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  egrep: warning: egrep is obsolescent; using grep -E
   Initialized empty Git repository in ...
   
   $ fn_git_commit -m 'rename and add submodule'

ERROR: test-renames.t output changed
```

Reply:
Could you send the patch upstream? If not, I can do it later today.

Reply:
I made [an issue](https://foss.heptapod.net/mercurial/hg-git/-/issues/400) first asking if they will be accepting patch for this.